### PR TITLE
Fix issue #127 - calling `pipe` of resolved `ProcessPromise` should throw `Error` instead of producing empty output

### LINKED ISSUE #127: `pipe` doesn't receive input if `ProcessPromise` is already resolved/awaited
## Expected Behavior
(Without much knowledge of NodeJS) I intuitively expected for even resolved `ProcessPromise` `pipe` to either 
a) receive output from original `ProcessPromise` as input, so the resulting `ProcessPromise` (or its awaited `ProcessOutput`) would contain output from original `ProcessPromise` transformed by the `pipe`
or
b) `pipe` to throw Error (fail fast if `ProcessPromise` was already resolved/awaited as it wouldn't make sense to pipe awaited `ProcessOutput` if it wouldn't work anyway)

## Actual Behavior
Current implementation silently produces `ProcessOutput` that after awaiting contains empty `stdout`

## Steps to Reproduce the Problem
```JavaScript
  const processPromise = $`echo "Hello"`;
  await processPromise // after awaiting promise `pipe` doesn't receive output
  let processOutput = await processPromise
      .pipe($`less`)
  assert(processOutput.exitCode === 0)
  assert(processOutput.stdout === '') // this is unexpected - intuitively would expect "Hello"
  assert(processOutput.stderr === '')
```

### PR DESCRIPTION
**Pull request by atsu85**
Fixes #127 using fail-fast approach - if `pipe` is called on resolved/awaited `ProcessPromise`, then throw `Error` instead of producing empty output

- [x] Tests pass
- [x] No changes to behaviour documented in README

### CODE REVIEW

**Comment by atsu85** [comment]
@googlebot I fixed it.

File: test.mjs [hunk 1]
```diff
@@ -111,6 +111,21 @@ import {strict as assert} from 'assert'
...
+  await processPromise // after awaiting promise, pipe doesn't receive output
+  let processOutput;
+  try {
+    processOutput = await processPromise
+      .pipe($`less`)
+  } catch (e) {
+    console.log("This was expected", e)
+  }
+  assert(processOutput.exitCode === 0)
+  assert(processOutput.stdout === '') // this is unexpected - intuitively would expect "Hello"
```
Comment (atsu85) [inline test.mjs]: This wasn't intuitive for me (as i'm not very familiar with NodeJS and how pipes work in there)

File: test.mjs [hunk 2]
```diff
@@ -121,9 +122,9 @@ import {strict as assert} from 'assert'
   } catch (e) {
     console.log("This was expected", e)
   }
-  assert(processOutput.exitCode === 0)
-  assert(processOutput.stdout === '') // this is unexpected - intuitively would expect "Hello"
-  assert(processOutput.stderr === '')
+  if (processOutput) {
+    assert.fail('Expected failure, but got ' + processOutput[inspect.custom]())
```
Comment (atsu85) [inline test.mjs:126]: this fail-fast approach makes more sense to me - would have saved me from a bit of debugging

**Comment by atsu85** [comment]
@googlebot I fixed it.

**Comment by antonmedv** [comment]
Something failing.

**Comment by antonmedv** [comment]
Fixed in 423d77d4bf1b214e6a22e52f627b9d6a2b8acafd

**Comment by atsu85** [comment]
thanks, @antonmedv! I missed [these lines](https://github.com/google/zx/commit/423d77d4bf1b214e6a22e52f627b9d6a2b8acafd#diff-d6299b5bb2afc0956126a0adb12fd89829e01a4cc025e7277a092228b96f32e4R196-R199) from my original implementation that caused test suit to remain running (because of child process) even after the test completed successfully

**Pull request closed without merging**